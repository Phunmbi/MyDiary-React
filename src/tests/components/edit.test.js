import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import EditComponent, { Edit } from '../../components/Edit';
import Root from '../../root';

let wrapper;
let wrapped;

document.getElementById = jest.fn((selector) => {
  console.log('selector => ', selector);

  return {
    value: {
      data: {
        title: 'value',
      },
    },
    style: {
      background: 'background',
    },
    innerHTML: 'inner',
    disabled: true,
  };
});

const props = {
  viewEntry: jest.fn(),
  editEntry: jest.fn(),
  response: {
    singleEntry: {
      status: 200,
      data: {
        title: 'title',
      },
    },
    editedEntry: {
      response: {
        status: 200,
      },
    },
  },
  match: {
    params: {
      id: 1,
    },
  },
};

beforeEach(() => {
  wrapper = mount(
    <Root>
      <MemoryRouter initialEntries={[{ key: 'testkey' }]}>
        <EditComponent
          match={props.match}
          response={props.response}
          viewEntry={props.viewEntry}
        />
      </MemoryRouter>
    </Root>,
  );

  wrapped = shallow(
    <Edit
      match={props.match}
      response={props.response}
      viewEntry={props.viewEntry}
    />,
  );
});

afterEach(() => wrapper.unmount());

describe('Edit UI', () => {
  describe('render features', () => {
    test('component should render as expected', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });
});

describe('Edit Functionality', () => {
  it('should click simulate edit', () => {
    const submitButton = wrapped.find('form');

    const inst = wrapped.instance();

    inst.props = {
      editEntry: jest.fn(),
      match: {
        params: {
          id: 1,
        },
      },
    };

    const mockedEvent = {
      preventDefault: jest.fn(),
      target: {
        reset: jest.fn(),
        children: {
          title: 'title',
          details: 'details',
          submit: {
            innerHTML: 'submit',
            style: {
              background: 'background',
            },
          },
        },
      },
    };
    submitButton.simulate('submit', mockedEvent);
  });

  it('should click simulate edit 2', () => {
    const inst = wrapped.instance();

    inst.props = {
      editEntry: jest.fn(),
      match: {
        params: {
          id: 1,
        },
      },
    };

    wrapped.setState({});
  });
});
