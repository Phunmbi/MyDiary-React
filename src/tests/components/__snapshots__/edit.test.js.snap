// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit UI render features component should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testkey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testkey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testkey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(Edit)
          match={
            Object {
              "params": Object {
                "id": 1,
              },
            }
          }
          response={
            Object {
              "editedEntry": Object {
                "response": Object {
                  "status": 200,
                },
              },
              "singleEntry": Object {
                "data": Object {
                  "title": "title",
                },
                "status": 200,
              },
            }
          }
          viewEntry={
            [MockFunction] {
              "calls": Array [
                Array [
                  1,
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
        >
          <Edit
            editEntry={[Function]}
            match={
              Object {
                "params": Object {
                  "id": 1,
                },
              }
            }
            response={
              Object {
                "authenticated": "",
                "deletedEntry": Array [],
                "deletedEntryError": Array [],
                "deletedReminder": Array [],
                "deletedReminderError": Array [],
                "editedEntry": Array [],
                "editedEntryError": Array [],
                "entries": Array [],
                "entriesError": Array [],
                "error": Array [],
                "existingReminder": Array [],
                "existingReminderError": Array [],
                "firstName": "",
                "lastName": "",
                "newEntry": Array [],
                "newEntryError": Array [],
                "newReminder": Array [],
                "newReminderError": Array [],
                "singleEntry": Array [],
                "singleEntryError": Array [],
                "status": "",
              }
            }
            viewEntry={[Function]}
          >
            <div
              className="dashboard"
            >
              <Header>
                <header>
                  <Link
                    replace={false}
                    to="/dashboard"
                  >
                    <a
                      href="/dashboard"
                      onClick={[Function]}
                    >
                      <h1>
                        MyDiary
                      </h1>
                    </a>
                  </Link>
                  <Link
                    replace={false}
                    to="/profile"
                  >
                    <a
                      href="/profile"
                      onClick={[Function]}
                    >
                      <img
                        alt="Profile"
                        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+P+/HgAFhAJ/wlseKgAAAABJRU5ErkJggg=="
                      />
                    </a>
                  </Link>
                </header>
              </Header>
              <div
                id="loadingModal"
              >
                <div
                  className="loadingModal-main"
                >
                  <img
                    alt="Loading"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAEBAQAAACH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  />
                </div>
              </div>
            </div>
          </Edit>
        </Connect(Edit)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;
